_posts/2020-01-12-updating-foreman-inventory-with-system-facts.md

```diff
@@ -0,0 +1,311 @@
+
+Once facts are imported the parsing follows. We normalize them and update Host attributes. For example, Puppet reports uptime while subscription-manager reports boot time timestamp. At this point we calculate the boot time and store that in a unified form which can be easily used for searching in Foreman inventory.
+
+We also update the host attributes, such as the operating system or even create a new OS object if it does not exist yet. As mentioned earlier, if we see facts for the host that does not exist in the Foreman database yet, Foreman can create it. This can be used to populate the inventory easily when you start with Foreman in the brownfield deployment.
+
+To customize the fact importing behavior, the following settings can be tweaked. Note that they
+apply to every fact source, despite the fact, some of them are listed under
+**Puppet** and **Provisioning** settings category.
+
+<table class="table table-bordered table-condensed">
+  <thead>
+    <tr>
+      <th>Setting</th>
+      <th>Category</th>
+      <th>Effect</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>Create new host when facts are uploaded</td>
+      <td>Puppet</td>
+      <td>If set to true (default), if facts for an unknown hosts are received, Host object will be created based on them</td>
+    </tr>
+    <tr>
+      <td>Create new host when report is uploaded</td>
+      <td>Puppet</td>
+      <td>If set to true (default), if a report for an unknown host is received, Host object will be created based on it. Report typically contains less information. Report usually contains information about what has happened during the configuration deployment, but at least the host name is available</td>
+    </tr>
+    <tr>
+      <td>Default location</td>
+      <td>Puppet</td>
+      <td>When host is created based on either facts, we search for a fact representing the location. If there's no such fact or the host  is created based on the report, it will be assigned to the location specified in this setting.</td>
+    </tr>
+    <tr>
+      <td>Default organization</td>
+      <td>Puppet</td>
+      <td>Same as default location, but works for organization</td>
+    </tr>
+    <tr>
+      <td>Location fact</td>
+      <td>Puppet</td>
+      <td>The name of the fact, which contains the name of the location, in which this host should be placed. By default, there's no such  fact, but all configuration management systems allow to specify custom fact. Hence you can create a custom fact e.g. foreman_location and then the Host will be automatically assigned to it. </td>
+    </tr>
+    <tr>
+      <td>Organization fact</td>
+      <td>Puppet</td>
+      <td>Same as above but for organization</td>
+    </tr>
+    <tr>
+      <td>Update subnets from facts</td>
+      <td>Puppet</td>
+      <td>If an IP of existing interface is changed (or newly created), Foreman will try to match existing Subnet object based on this setting.</td>
+    </tr>
```

Comment:
Will it create the subnet as well? I don't remember.

Reply:
I don't so, we don't get enough information, only domains are auto-created, I'll double check and will add that information here